### ISSUE TITLE
New sections_url_name is not backwards-compatible

### ISSUE THREAD

**Issue by czlee** [EliotBerriot/django-dynamic-preferences#67]
Hi! This isn't too big a deal but I thought I'd mention in case you weren't aware and wanted to include it in your documentation somewhere: 0.8.3 isn't backwards-compatible with existing projects using 0.8.2, because it requires a new class property `sections_url_name` (in 723f2ece8376b07bbb9ccb5ee689a4fca08ae20f). I've implemented it on our side (czlee/tabbycat#397) so no worries in terms of functionality for us, just wanted to raise a flag 😃 

Thanks again for all your work on this app, it's super helpful for our project!

**Comment by EliotBerriot** [comment]
Thank you for opening this. Backward compatibility is a serious issue, and I'll mark the 0.8.3 release as not backward compatible (as well as publishing a new release to revert this patch).

If I remember well, this was introduced so the template displaying preferences forms could also display link to specific sections. It's not really important nor a core feature (or at least, it's *less* important than backward compatibility).

**Comment by EliotBerriot** [comment]
The 0.8.4 release was just pushed on PyPi to address this. I'm really sorry this happened. Thanks again for reporting this.

**Comment by czlee** [comment]
That's excellent, thanks so much for getting to it so quickly!

I thought I should perhaps draw attention to the fact that including the `sections_url_name` setting now crashes 0.8.4, so 0.8.4 is also not backwards-compatible with 0.8.3. This is again fine with us—we'll just drop the `sections_url_name` attribute and set our pip requirements file to insist on having at least 0.8.4—but just wanted to check this was what you intended? As opposed to, say, having 0.8.4 silently ignore a `sections_url_name` attribute.

To be more specific—of course in general extraneous attributes are ignored in Python, but for convenience we're instantiating `PreferenceFormView` subclasses using the `.as_view(sections_url_name='url-name')` method, which throws a `TypeError` if you use a keyword argument that isn't already an attribute of the class it's instantiating. (See czlee/tabbycat@bb6544d53987d2fd60543b4901566603eb0a73bf.) So if you wanted it to be backwards-compatible with both 0.8.3 and 0.8.2, it might work just to include the `sections_url_name = None` line in `PreferenceFormView` (and nothing else).

Just a thought—whatever you decide will work well for our project. Thanks again!

**Comment by EliotBerriot** [comment]
After thinking about it I'd rather remove any tracks of this `sections_url_name`, even if it means the `0.8.4` is not backward compatible with `0.8.3`, since the latter is marked as a problematic release in the changelog itself. 

The timewindow to face this bug was less than a week, so my guess is that any user that faced the 0.8.3 bug will come here to see what's going on and upgrade to the latest version to fix the bug.

To sum it up, I'd rather not being backward compatible with a backward incompatible version when the backward incompatible version was backward incompatible by mistake and not by design.

To anybody reading this, feel free to contribute if you have a better idea.
